jib-core/src/main/java/com/google/cloud/tools/jib/ncache/Cache.java

```diff
@@ -16,22 +16,82 @@
 
 package com.google.cloud.tools.jib.ncache;
 
+import com.google.cloud.tools.jib.blob.Blob;
 import com.google.cloud.tools.jib.image.DescriptorDigest;
 import com.google.cloud.tools.jib.image.LayerEntry;
 import com.google.common.collect.ImmutableList;
 import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.NotDirectoryException;
+import java.nio.file.Path;
 import java.util.Optional;
+import javax.annotation.concurrent.Immutable;
 
-/** Cache for storing data to be shared between Jib executions. */
+/**
+ * Cache for storing data to be shared between Jib executions.
+ *
+ * <p>Uses the default cache storage engine ({@link DefaultCacheStorage}) and layer entries as the
+ * selector ({@link LayerEntriesSelector}).
+ *
+ * <p>Implementation is immutable and thread-safe.
+ */
+@Immutable
 public class Cache {
 
+  /**
+   * Initializes the cache using {@code cacheDirectory} for storage.
+   *
+   * @param cacheDirectory the directory for the cache. Creates the directory if it does not exist.
+   * @return a new {@link Cache}
+   * @throws NotDirectoryException if {@code cacheDirectory} is not a directory
+   * @throws IOException if an I/O exception occurs
+   */
+  public static Cache withDirectory(Path cacheDirectory) throws IOException {
+    if (Files.exists(cacheDirectory)) {
+      if (!Files.isDirectory(cacheDirectory)) {
+        throw new NotDirectoryException("The cache can only write to a directory");
```

Comment:
Not right or wrong, but createDirectories throws FileAlreadyExistsException in this case: https://docs.oracle.com/javase/7/docs/api/java/nio/file/Files.html#createDirectories(java.nio.file.Path,%20java.nio.file.attribute.FileAttribute...)

It also uses some "optional specific exception" hint in the java doc.

Reply:
My actual comment is, should we just use `FileAlreadyExistsException` here?

Reply:
Yea, that sounds good.